flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -909,6 +926,70 @@ private void clearDispatcherState() {
 		terminateJobManagerRunners();
 	}
 
+	private void instantiateJobManagerOverviewMetrics(MetricGroup jobManagerMetricGroup) {
+		long defaultTimeoutValue = configuration.getLong(RestOptions.CONNECTION_TIMEOUT);
+		Time defaultRequestTimeout = Time.milliseconds(defaultTimeoutValue);
+
+		jobManagerMetricGroup.gauge("taskSlotsAvailable", () -> {
+			try {
+				return (long) resourceManagerGateway
+					.requestResourceOverview(defaultRequestTimeout)
+					.get()
+					.getNumberFreeSlots();
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+			} catch (ExecutionException e) {
+				log.error("Request resource overview occurs exception.", e);
+			}
+
+			return 0L;
+		});
+
+		jobManagerMetricGroup.gauge("taskSlotsTotal", () -> {
+			try {
+				return (long) resourceManagerGateway
+					.requestResourceOverview(defaultRequestTimeout)
+					.get()
+					.getNumberRegisteredSlots();
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+			} catch (ExecutionException e) {
+				log.error("Request resource overview occurs exception.", e);
+			}
+
+			return 0L;
+		});
+
+		jobManagerMetricGroup.gauge("numRegisteredTaskManagers", () -> {
+			try {
+				return (long) resourceManagerGateway
+					.requestResourceOverview(defaultRequestTimeout)
+					.get()
+					.getNumberTaskManagers();
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+			} catch (ExecutionException e) {
+				log.error("Request resource overview occurs exception.", e);
+			}
+
+			return 0L;
+		});
+
+		jobManagerMetricGroup.gauge("numRunningJobs", () -> {
+			try {
+				return (long) requestOverviewForAllJobs(defaultRequestTimeout)
```

Comment:
This results in a wave of RPC calls to every single JM. Would it not be feasible to simple count the number of job IDs contained in `jobManagerRunnerFutures`?